doc/contrib/new_api.md

```diff
@@ -0,0 +1,510 @@
+# PREVIEW: New `libelektra-core` API 2023
+
+## Core API
+
+### Note on names
+
+From the [use cases](/doc/usecases/core/README.md) we know that there is a collection structure which contains elements with a name and value (and metadata).
+
+The current API uses these names for those concepts:
+
+- Collection: `KeySet`
+- Element: `Key`
+- Name of element: —
+- Value of element: —
+
+The API proposed below uses (to keep as much as possible from above):
+
+- Collection: `ElektraKeyset`
+- Element: `ElektraKey`
+- Name of element: `ElektraKeyname`
+- Value of element: `ElektraKeyvalue`
+
+Another (maybe less confusing) option would be:
+
+- Collection: `ElektraSet` or `ElektraMap`
+- Element: `ElektraEntry`
+- Name of element: `ElektraKey` or `ElektraName`
+- Value of element: `ElektraValue`
```

Comment:
I don't think this is a good idea. This would need a convincing usecase.

Reply:
I don't quite understand the problem here... Basically what this say is:
1. The term for "the value of an Element" is: _value_
2. If we need to represent it as a type it is: `ElektraValue`

We might not use `ElektraValue` in C (let's discuss that in the structs section below), but in some bindings it could make sense. For example in Haskell:

```haskell
module Elektra (
  Value
) where

import Data.ByteString

newtype Value = Data.ByteString
```

Reply:
Again. This part here is not about introducing `ElektraValue` as a type or identifier **anywhere**. This text is basically just a preview for the glossary. It just says we call this thing "value" (no other special name). The `ElektraValue` part basically just says **if** we found a reason to introduce a type, `ElektraValue` would be time. Essentially we reserve the name for this purpose.